newrelic/api/import_hook.py

```diff
@@ -173,25 +168,16 @@ def find_module(self, fullname, path=None):
         self._skip[fullname] = True
 
         try:
-            # For Python 3 we need to use find_loader() from the new
-            # importlib module.
-
-            if find_loader:
-                loader = find_loader(fullname, path)
-
-                if loader:
-                    return _ImportHookChainedLoader(loader)
-
-            else:
-                __import__(fullname)
-
-                # If we get this far then the module we are
-                # interested in does actually exist and so return
-                # our loader to trigger import hooks and then return
-                # the module.
-
-                return _ImportHookLoader()
-
+            try:
+                from importlib.util import find_spec
+                return _ImportHookChainedLoader(find_spec(fullname).loader)
+            except ImportError:
+                try:
+                    from importlib import find_loader
+                    return _ImportHookChainedLoader(find_loader(fullname, path))
```

Comment:
This is a change in behavior, the previous implementation returned `None` for the case where `find_loader` returned `None`.

Reply:
Nice catch. This branch (the <=py3.4 fallback) is now completely removed, since the newrelic python agent doesn't run on <=py3.4, and is therefore never run anyway.